app/src/main/java/com/mlinde/marvelcomicsapp/comicsList/ComicsListViewModel.kt

```diff
@@ -14,19 +15,14 @@ import javax.inject.Inject
 class ComicsListViewModel @Inject constructor(private val repository: ComicsRepository): ViewModel() {
 
     var comicsLiveData: MutableLiveData<ComicDataWrapper> = MutableLiveData()
-    var messageLiveData: MutableLiveData<String> = MutableLiveData()
 
     fun getComics(){
         viewModelScope.launch {
-            when(val response = repository.getComics()){
-                is ApiRensponse.Success ->{
-                    comicsLiveData.postValue(response.data as ComicDataWrapper)
+            runCatching { repository.getComics() }
+                .onSuccess { comicsLiveData.postValue(it.body()) }
```

Comment:
It's ok. But if you want to add more states for example Loading State to your `ApiResponse`, then better way is to have `ApiResponse<ComicDataWrapper>` as object in LiveData.